dask/bag/core.py

```diff
@@ -817,16 +817,34 @@ def topk(self, k, key=None, split_every=None):
         return self.reduction(func, compose(func, toolz.concat), out_type=Bag,
                               split_every=split_every, name='topk')
 
-    def distinct(self):
+    def distinct(self, key=None):
         """ Distinct elements of collection
 
         Unordered without repeats.
 
+        Parameters
+        ----------
+        key: {callable,str}
+            Defines uniqueness of items in bag by calling `key` on each item.
+            If a string is passed `key` is considered to be `lambda x: x[key]`.
```

Comment:
```suggestion
            If a string is passed ``key`` is considered to be ``lambda x: x[key]``.
```

Our docstrings are rendered by sphinx, and so need to follow restructured-text formatting, for which double-backticks are used for quotes.